encode_l2.go

```
@@ -106,12 +106,25 @@ func encodeBlockBetterGo(dst, src []byte) (d int) {
+	// region and loses the matches there. The assembly has clamped since it was
+	// written ("Blocks can be long, limit max skipping"); the Go path had not,
+	// which is why the two produced different output on blocks above 64KB.
+	//
+	// Only this encoder clamps. encodeBlockBetterGo64K mirrors
+	// encodeBetterBlockAsm64K, which does not clamp -- and the dispatch
+	// boundary between the two is exactly the 64KB one used here, so the
+	// pairing is exact on both sides.
+	const maxSkip = 100
+
 	for {
 		candidateL := 0
```

golangci-lint (2.12.2)

[error] assigned to candidateL, but reassigned without using the value

(wastedassign)